packages/ui/src/components/form-input/components/form-textarea.tsx

```diff
@@ -0,0 +1,33 @@
+import { forwardRef } from 'react'
+import { Controller, useFormContext } from 'react-hook-form'
+
+import { Textarea, TextareaProps } from '@components/form-primitives'
+
+interface FormTextareaPropsType extends TextareaProps {
+  name: string
+}
+
+const FormTextarea = forwardRef<HTMLTextAreaElement, FormTextareaPropsType>((props, ref) => {
+  const formContext = useFormContext()
+
+  if (!formContext) {
+    throw new Error(
+      'FormTextarea must be used within a FormProvider context through FormWrapper. Use the standalone Textarea component if form integration is not required.'
+    )
+  }
+
+  return (
+    <Controller
+      name={props.name}
+      control={formContext.control}
+      render={({ field, fieldState }) => {
+        const error = fieldState.error?.message || props.error
+        return <Textarea {...props} {...field} error={error} ref={ref} theme={error ? 'danger' : 'default'} />
```

Comment:
We can worry about `theme` in the `Textarea` component
```suggestion
        return <Textarea {...props} {...field} error={fieldState.error?.message || props.error} ref={ref} />
```

Reply:
Sorry, updated